spring-boot-project/spring-boot-test/src/main/java/org/springframework/boot/test/context/TestSliceTestContextBootstrapper.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2012-2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.test.context;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Method;
+
+import org.springframework.core.ResolvableType;
+import org.springframework.test.context.TestContextAnnotationUtils;
+import org.springframework.test.context.TestContextBootstrapper;
+import org.springframework.util.Assert;
+
+/**
+ * {@link TestContextBootstrapper} for test slices support.
+ *
+ * @param <T> the test slice annotation
+ * @author Yanming Zhou
+ * @since 3.5.0
+ */
+public class TestSliceTestContextBootstrapper<T extends Annotation> extends SpringBootTestContextBootstrapper {
+
+	private final Class<T> annotationType;
+
+	private final Method propertiesMethod;
+
+	@SuppressWarnings("unchecked")
+	public TestSliceTestContextBootstrapper() {
+		this.annotationType = (Class<T>) ResolvableType.forClass(getClass())
+			.as(TestSliceTestContextBootstrapper.class)
+			.getGeneric(0)
+			.resolve();
+		Assert.notNull(this.annotationType, "'%s' doesn't contain type parameter of '%s'"
+			.formatted(getClass().getName(), TestSliceTestContextBootstrapper.class.getName()));
+		try {
+			this.propertiesMethod = this.annotationType.getDeclaredMethod("properties");
+			Assert.state(this.propertiesMethod.getReturnType() == String[].class,
+					"Method '%s.properties()' should return String[]".formatted(this.annotationType.getName()));
+		}
+		catch (NoSuchMethodException ex) {
+			throw new RuntimeException(
```

Comment:
I need advice for exception type and message.

Reply:
You can simplify things a little bit

```java
public class TestSliceTestContextBootstrapper<A extends Annotation> extends SpringBootTestContextBootstrapper {

	private final Class<A> annotationType;


	@Override
	protected String[] getProperties(Class<?> testClass) {
		MergedAnnotation<A> annotation = MergedAnnotations.search(SearchStrategy.TYPE_HIERARCHY)
			.withEnclosingClasses(TestContextAnnotationUtils::searchEnclosingClass)
			.from(testClass)
			.get(this.annotationType);
		return annotation.getStringArray("properties");
	}

}
```

Reply:
Yes, I have considered this way, but I think it's better to keep the logic in `TestContextAnnotationUtils`, WDYT?

Reply:
> but I think it's better to keep the logic in TestContextAnnotationUtils, WDYT?

Honestly, I don't see any issues with using `MergedAnnotations.search(...)` instead of
`TestContextAnnotationUtils`. On one hand, it would be nice if the Spring Framework
could provide this functionality, but it does not make the situation significantly better.
Considering that it is used only once in a single place, IMO, it is too much to ask to add it to the Spring Framework.

Also, this PR is still in a _draft_ state, and there is no feedback from the Spring Boot team regarding
this enhancement, so maybe it makes sense to wait for feedback first and then ask the Spring Framework team to
add something.

Reply:
Updated as you suggested, thank you @nosan